### ISSUE TITLE
Can sometimes be worse than pure Muon (and Adam)

### ISSUE THREAD

**Issue by fgdfgfthgr-fox** [Chongjie-Si/AdaMuon#5]
Settings:
A [DiP](https://github.com/NJU-PCALab/DiP) image transformer.
On tanganke/resisc45 dataset.
Class-conditioned, flow matching (v-loss) model.
Works on 256x256 resolution, and a patch size of 16.
Rope for positional encoding.
384 hidden dimension and 8 layers. Total ~16M parameters (it's small because just prototyping).
The quality of validation is measured by FD-DINOv2 score. (lower is better)

Reference is AdamW with 2e-4 learning rate, no weight decay.
Uses [pytorch-optimizer](https://pypi.org/project/pytorch_optimizer/) for AdaMuon implementation. I had make sure only 2D parameters within DiT's linear layer are using AdaMuon. While token embeddings and convolutions uses AdamW. The AdamW part of AdaMuon (and Muon) uses 2e-4 lr.

AdaMuon paper claims "...to match the RMS norm of Muon’s updates to that of Adam, thereby enabling a unified learning rate schedule..." and "Notably, AdaMuon consistently achieves the highest efficiency, regardless of model size or learning rate".

At 2e-4 lr, AdaMuon performs much worse than AdamW. FD-DINOv2 score consistently higher than AdamW at all stage of training, and stops decreasing after around 40k steps while AdamW see continues decrease.

<img width="940" height="380" alt="Image" src="https://github.com/user-attachments/assets/d333f3da-b80f-428c-97b0-c3d94597954c" />
(Cyan = AdaMuon, Orange = AdamW).

At 2e-3 lr, AdaMuon performs somewhat better than AdamW:

<img width="925" height="368" alt="Image" src="https://github.com/user-attachments/assets/3b595998-d22d-4737-8166-5e3d8d99e52a" />
(Red = AdaMuon, Orange = AdamW).

However, also at 2e-3 lr, Muon performs way better (training still early but I can see the trend).

<img width="925" height="366" alt="Image" src="https://github.com/user-attachments/assets/8e2d45b1-a724-406f-be9c-e149c7098477" />

(Red = AdaMuon, Orange = Muon).

Did I misunderstand something about AdaMuon? I can provide code if needed.
